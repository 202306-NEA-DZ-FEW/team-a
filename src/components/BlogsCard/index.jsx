import Image from "next/image";
import Link from "next/link";
import { useTranslation } from "next-i18next";

import { formatDate } from "@/lib/helpers";

export default function BlogsCard({ index, blog }) {
  const { i18n, t } = useTranslation();
  const date = formatDate(blog.createdAt);
  return (
    <>
      <div
        className={`flex flex-col gap-10 justify-center items-center  ${
          index % 2 === 0 ? "lg:flex-row-reverse" : "lg:flex-row"
        } `}
        dir={i18n?.language === "ar" ? "rtl" : "ltr"}
      >
        <div className='relative lg:flex-1'>
          <figure className='relative w-80 h-60 lg:w-[100%] lg:h-80'>
            <Image
              src={blog.imageUrl}
              alt={blog.title}
              fill
              priority
              sizes='(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw'
              className='object-cover w-full h-full rounded-2xl'
            />
          </figure>

          <span className='absolute bottom-0 left-0 bg-primary text-white p-2 text-sm lg:hidden'>
            {date}
          </span>
        </div>
        <div className='flex flex-col lg:flex-1 gap-4'>
          <div>
            <h2 className='font-bold text-2xl'>{blog.title}</h2>
            <span className='my-1 text-sm text-gray-500 font-light hidden lg:block'>
              {date}
            </span>
          </div>

          <p className='text-slate-500'>
            {blog.description.slice(0, 210)}
            {blog.description.length > 210 ? " ..." : ""}
          </p>
          <Link href={`/blogs/${blog.id}`}>
            <div>
              <button className='btn bg-black text-white btn-sm'>
                {t("common:buttons:readMore")}
              </button>
            </div>
          </Link>
        </div>
      </div>
    </>
  );
}
