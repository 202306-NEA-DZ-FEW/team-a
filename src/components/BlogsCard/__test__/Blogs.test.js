import renderer from "react-test-renderer";

import BlogsCard from "..";

const testBlog = {
  id: 1,
  title: "Sample Blog Title",
  imageUrl: "/sample-image.jpg",
  description: "Sample description for the blog...",
  date: "2023-10-28",
};
it("renders correctly", () => {
  const tree = renderer.create(<BlogsCard blog={testBlog} />).toJSON();
  expect(tree).toMatchSnapshot();
});
