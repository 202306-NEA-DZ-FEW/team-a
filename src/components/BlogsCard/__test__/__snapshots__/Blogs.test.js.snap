// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex flex-col gap-10 justify-center items-center  lg:flex-row "
  dir="ltr"
>
  <div
    className="relative lg:flex-1"
  >
    <figure
      className="relative w-80 h-60 lg:w-[100%] lg:h-80"
    >
      <img
        alt="Sample Blog Title"
        className="object-cover w-full h-full rounded-2xl"
        data-nimg="fill"
        decoding="async"
        fetchpriority="high"
        onError={[Function]}
        onLoad={[Function]}
        sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
        src="/_next/image?url=%2Fsample-image.jpg&w=3840&q=75"
        srcSet="/_next/image?url=%2Fsample-image.jpg&w=256&q=75 256w, /_next/image?url=%2Fsample-image.jpg&w=384&q=75 384w, /_next/image?url=%2Fsample-image.jpg&w=640&q=75 640w, /_next/image?url=%2Fsample-image.jpg&w=750&q=75 750w, /_next/image?url=%2Fsample-image.jpg&w=828&q=75 828w, /_next/image?url=%2Fsample-image.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fsample-image.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fsample-image.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fsample-image.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fsample-image.jpg&w=3840&q=75 3840w"
        style={
          Object {
            "bottom": 0,
            "color": "transparent",
            "height": "100%",
            "left": 0,
            "objectFit": undefined,
            "objectPosition": undefined,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": "100%",
          }
        }
      />
    </figure>
    <span
      className="absolute bottom-0 left-0 bg-primary text-white p-2 text-sm lg:hidden"
    >
      Invalid date
    </span>
  </div>
  <div
    className="flex flex-col lg:flex-1 gap-4"
  >
    <div>
      <h2
        className="font-bold text-2xl"
      >
        Sample Blog Title
      </h2>
      <span
        className="my-1 text-sm text-gray-600 font-light hidden lg:block"
      >
        Invalid date
      </span>
    </div>
    <p>
      Sample description for the blog...
    </p>
    <a
      href="/blogs/1"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <div>
        <button
          className="btn bg-black text-white btn-sm"
        >
          buttons.readMore
        </button>
      </div>
    </a>
  </div>
</div>
`;
